src/Illuminate/Foundation/Providers/FoundationServiceProvider.php

```diff
@@ -283,11 +282,12 @@ protected function registerExceptionRenderer()
      */
     public function registerMaintenanceModeManager()
     {
-        $this->app->singleton(MaintenanceModeManager::class);
+        $this->app->singleton('maintenance.manager', function ($app) {
+            return new MaintenanceModeManager($app);
+        });
```

Comment:
You don't need to assign it a string name that is different than the class name. This could be a breaking change. You can just keep using the class name as the binding name.

Reply:
@taylorotwell I was matching up how other managers and drivers are bound in the container here. The core aliases in `Application` above ensure class names still work:

**Before:**
```
dump(resolve(\Illuminate\Foundation\MaintenanceModeManager::class)::class); // Illuminate\Foundation\MaintenanceModeManager
dump(resolve(\Illuminate\Contracts\Foundation\MaintenanceMode::class)::class); // Illuminate\Foundation\FileBasedMaintenanceMode
```

**After:**
```
dump(resolve(\Illuminate\Foundation\MaintenanceModeManager::class)::class); // Illuminate\Foundation\MaintenanceModeManager
dump(resolve(\Illuminate\Contracts\Foundation\MaintenanceMode::class)::class); // Illuminate\Foundation\FileBasedMaintenanceMode
```

Perhaps I've missed something though?